### PR TITLE
fix(schema): fix the typehint for nested extractors

### PR DESCRIPTION
**Pull request by Lee-000**

### CODE REVIEW

**Comment by GravitiRob** [comment]
Task linked: [CU-2cy5uvx 确定 field name + type 到 extractor + dtype 映射的格式，并实现 filename 作为 example](https://app.clickup.com/t/2cy5uvx)

File: graviti/schema/extractors.py [hunk 1]
```diff
@@ -17,7 +17,16 @@
 _T = TypeVar("_T")
 _D = Dict[str, Tuple[Iterator[_T], str]]
 _Extractor = Tuple[Callable[[Dict[str, Any]], Iterator[_T]], str]
-_Extractors = Dict[str, _Extractor[_T]]
+
+
+class Extractors(Protocol, Generic[_T]):
```
Comment (linjiX) [inline graviti/schema/extractors.py:22]: Why we need `Generic`?

Comment (linjiX) [reply graviti/schema/extractors.py:22]: Will a private class better?

**Review by linjiX — CHANGES_REQUESTED** [review]

File: graviti/schema/extractors.py (hunk 1 above)
Comment (graczhual) [reply graviti/schema/extractors.py:22]: > Will a private class better?

This class will be used in lazy.py

Comment (linjiX) [reply graviti/schema/extractors.py:22]: Other types around it are all private?

Comment (linjiX) [reply graviti/schema/extractors.py:22]: > > Why we need `Generic`?
> 
> To specify the type of the values in `Iterator`.

But all _T in `Extractors` is `Any`?

File: graviti/schema/extractors.py [hunk 2]
```diff
@@ -17,7 +17,16 @@
 _T = TypeVar("_T")
 _D = Dict[str, Tuple[Iterator[_T], str]]
 _Extractor = Tuple[Callable[[Dict[str, Any]], Iterator[_T]], str]
-_Extractors = Dict[str, _Extractor[_T]]
+
```
Comment (linjiX) [inline graviti/schema/extractors.py:20]: Why not just set `Any` in `_Extractor`

**Review by graczhual — APPROVED** [review]

**Pull request merged**